src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -197,8 +198,38 @@ public void logBatch(String batchTypeName, List<Object> queryOrIdList, List<List
         }
     }
 
-    public void reloadFilters()
+    public synchronized void disableAuditLog()
+    {
+        logger.info("Audit logging is disabled, stopping any existing loggers"); //debug
+        if (this.auditLogger != null)
+        {
+            this.auditLogger.stop();
+            this.auditLogger = null;
+        }
+    }
+
+    public synchronized void enableAuditLog(AuditLogOptions auditLogOptions) throws ConfigurationException
     {
-        filter = AuditLogFilter.create();
+        logger.trace("Audit logging is being enabled. Reloading AuditLogOptions.");
+        IAuditLogger oldLogger = this.auditLogger;
+        filter = AuditLogFilter.create(auditLogOptions);
+
+        if (oldLogger != null && oldLogger.getClass().getSimpleName().equals(auditLogOptions.logger))
+        {
+            logger.info("New AuditLogger [{}] is same as existing logger, hence not initializing the logger", auditLogOptions.logger);
```

Comment:
Might want to make it clear that filters were still updated. Otherwise as a user they might think "oh it didn't do anything" when it did in fact reload the filters.

Reply:
👍

Reply:
StorageService:enableAuditLog() method logs the changes in audit log filters anyways. That would make it clear that even thought logger is not reloaded, filters are.